app/controllers/comments_controller.rb

```diff
@@ -1,11 +1,14 @@
 class CommentsController < ApplicationController
   def new
+    @tweet = Tweet.find(params[:tweet_id])
   end
 
   def create
-    @comment = current_user.comments.build(tweet_id: params[:id], text: params[:text])
+    if @tweet = Tweet.find(params[:tweet_id])
+      @comment = current_user.comments.build(tweet_id: params[:tweet_id], text: params[:text])
```

Comment:
バグに気づきづらいので、build時になるべくparamsを参照しないほうがよいでしょう。

```suggestion
      @comment = current_user.comments.build(tweet_id: @tweet.id, text: params[:text])
```